samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -60,29 +97,23 @@ public LocalityManager(CoordinatorStreamManager coordinatorStreamManager) {
    * @return the map of containerId: (hostname, jmxAddress, jmxTunnelAddress)
    */
   public Map<String, Map<String, String>> readContainerLocality() {
-    if (coordinatorStreamManager == null) {
-      throw new IllegalStateException("No coordinator stream manager to read container locality from.");
-    }
-
     Map<String, Map<String, String>> allMappings = new HashMap<>();
-    for (CoordinatorStreamMessage message : coordinatorStreamManager.getBootstrappedStream(
-        SetContainerHostMapping.TYPE)) {
-      SetContainerHostMapping mapping = new SetContainerHostMapping(message);
-      Map<String, String> localityMappings = new HashMap<>();
-      localityMappings.put(SetContainerHostMapping.HOST_KEY, mapping.getHostLocality());
-      localityMappings.put(SetContainerHostMapping.JMX_URL_KEY, mapping.getJmxUrl());
-      localityMappings.put(SetContainerHostMapping.JMX_TUNNELING_URL_KEY, mapping.getJmxTunnelingUrl());
-      allMappings.put(mapping.getKey(), localityMappings);
-    }
-    containerToHostMapping = Collections.unmodifiableMap(allMappings);
-
+    metadataStore.all().forEach((keyBytes, valueBytes) -> {
+        if (valueBytes != null) {
+          String locationId = valueSerde.fromBytes(valueBytes);
+          allMappings.put(keySerde.fromBytes(keyBytes), ImmutableMap.of(SetContainerHostMapping.HOST_KEY, locationId,
+                                                                        SetContainerHostMapping.JMX_TUNNELING_URL_KEY, "",
+                                                                        SetContainerHostMapping.JMX_URL_KEY, ""));
+        }
+      });
+    containerToHostMapping = allMappings;
```

Comment:
why is this no longer an `unmodifiableMap`? I don't see any places where you write a value to this map.

Reply:
It is, in general, a good practice to enforce immutability where ever possible especially, in concurrent programming. `containerToHostMapping` is only set in `readContainerLocality()` and not modified anywhere else. 

The other reason I feel strongly about this change is, your PR doesn't introduce any changes to the map as well. so why modify its properties? It makes it hard to backtrack and rule out changes during debugging.